posts/2021-02-15-eclipse-transformer.adoc

```diff
@@ -0,0 +1,311 @@
+
+Two specific updates are required when transforming test artifacts.  First, package references are updated, changing Java / Jakarta EE "javax" package references into Jakarta EE 9 "jakarta" package references.  Package references can be in several locations, including test applications, test code, and server configurations.  Second, server configurations are updated to replace Java / Jakarta EE features with corresponding Jakarta EE 9 features.  The Eclipse Transformer is used to perform the first of these updates.  Currently, updates to server configurations must be done manually.  (Server configuration features could be updated using the transformer, if rules data were created with the substitution values needed for this update.)
+
+== Invoking the Transformer
+
+The Eclipse Transformer can be used directly from the command line, as a build task, or from Java code.  This article describes using the transformer from the command line.
+
+(See the transformer project pages for detailed usage guidelines link:https://projects.eclipse.org/projects/technology.transformer[Eclipse Transformer].)
+
+=== Downloading the Transformer 
+
+Several downloads are available from the project pages link:https://projects.eclipse.org/projects/technology.transformer/downloads[Downloads].
+
+Current snapshots are available at link:https://oss.sonatype.org/content/repositories/snapshots/org/eclipse/transformer/org.eclipse.transformer.cli/[Snapshots].
+
+=== Building the transformer
+
+Alternatively, the transformer can be built by cloning the link:https://github.com/eclipse/transformer[Eclipse Transformer repository] and invoking maven to build the transformer jar:
+
+[source]
+--
+    maven clean package
+--
+
+=== Preparing a work area
+
+Download or copy the transformer binaries to a local directory:
+
+[source]
+--
+    mkdir <work>
+    cd <work>
+    cp transformer/org.eclipse.transformer.cli/target/org.eclipse.transformer.cli-0.3.0-SNAPSHOT-distribution.jar .
+    unzip org.eclipse.transformer.cli-0.3.0-SNAPSHOT-distribution.jar
+--
+
+Unpackaging the distribution obtains:
+
+[source]
+--
+    org.eclipse.transformer.cli-0.3.0-SNAPSHOT.jar
+    libs/org.eclipse.transformer-0.3.0-SNAPSHOT.jar
+    libs/biz.aQute.bnd.transform-5.2.0.jar
+    libs/slf4j-api-1.7.25.jar
+    libs/commons-cli-1.4.jar
+    libs/slf4j-simple-1.7.30.jar
+--
+
+=== Running the transformer with default rules
+
+The transformer may now be run using the default rules:
+
+[source]
+--
+  java -jar org.eclipse.transformer.cli-0.3.0-SNAPSHOT.jar \
+    <target_location> \
+    <output_location>
+--
+
+(The jar 'org.eclipse.transformer.cli-0.3.0-SNAPSHOT.jar' is runnable, with the invocation selecting as main class 'org.eclipse.transformer.jakarta.JakartaTransformer' and with the necessary jars from the 'libs' folder included in the class path.)
```

Comment:
I suggest something like:

(The 'org.eclipse.transformer.cli-0.3.0-SNAPSHOT.jar' file is a runnable jar file that specifies 'org.eclipse.transformer.jakarta.JakartaTransformer' as the main class and uses the necessary jars from the 'libs' folder in its class path.)

Reply:
Updated.